kinto/views/buckets.py

```diff
@@ -1,8 +1,57 @@
+from pyramid.httpexceptions import HTTPForbidden
+from pyramid.security import NO_PERMISSION_REQUIRED
+from pyramid.view import view_config
+
 from cliquet import resource
+from cliquet.utils import hmac_digest
+from cliquet.views.batch import build_request
 
 from kinto.views import NameGenerator
 
 
+@view_config(route_name='default_bucket', permission=NO_PERMISSION_REQUIRED)
+def default_bucket(request):
+    if (not hasattr(request, 'prefixed_userid') or
+            request.prefixed_userid is None):
+        raise HTTPForbidden  # Pass through the forbidden_view_config
+
+    settings = request.registry.settings
+    hmac_secret = settings['cliquet.userid_hmac_secret']
+    bucket_id = hmac_digest(hmac_secret, request.prefixed_userid)[:32]
```

Comment:
Can you explain what you do here ?

I don't see any spec for this in the tests :)